lib/model/query/entities.js

```diff
@@ -56,12 +58,67 @@ select ins.*, def.id as "entityDefId" from ins, def;`)
 
 // TODO: add and tests entity creation audit logging and error handling (failed validation, etc.)
 
+// HACK: This is a function to check if the entity *can* successfully be inserted
+// because without it... in the case of processing an entity within a worker,
+// the database hits its constraints and then breaks the rest of the transaction.
+// Then it won't do things like update the processing time on the relevant audit log event,
+// and I can't log any more errors :(
+const canInsertEntity = (datasetName, submissionDefId, entityUuid) => ({ one }) =>
```

Comment:
as discussed on slack, we can try `ON CONFLICT do something` to get rid of this function in another PR